backend/app/main.py

```diff
@@ -0,0 +1,147 @@
+from typing import Optional
+
+import httpx
+import jwt
+from authlib.common.urls import add_params_to_uri
+from authlib.integrations.starlette_client import OAuth, OAuthError
+from fastapi import Cookie, FastAPI, Query, Request
+from fastapi.responses import HTMLResponse, JSONResponse, RedirectResponse
+from fastapi.routing import APIRouter
+from httpx import Timeout
+from starlette.middleware.sessions import SessionMiddleware
+
+from .settings import conf
+
+app = FastAPI()
+app.add_middleware(SessionMiddleware, secret_key=conf.SESSION_SECRET)
+router = APIRouter()
+
+oauth = OAuth()
+oauth.register(
+    name="login_portal",
+    client_id=conf.OIDC_CLIENT_ID,
+    client_secret=conf.OIDC_CLIENT_SECRET,
+    server_metadata_url=(conf.OIDC_PROVIDER + "/.well-known/openid-configuration"),
+    client_kwargs={
+        "scope": conf.OIDC_SCOPES,
+        "timeout": Timeout(timeout=conf.OIDC_REQUEST_TIMEOUT),
+    },
+)
+
+
+@router.get("/login")
+async def login(request: Request):
+    return await oauth.login_portal.authorize_redirect(
+        request=request,
+        redirect_uri=request.url_for("auth"),
+        acr_values=conf.OIDC_ACR_VALUES,
+    )
+
+
+@router.get("/auth")
+async def auth(request: Request):
+    try:
+        token = await oauth.login_portal.authorize_access_token(request)
+    except OAuthError as error:
+        return HTMLResponse(f"<h1>{error.error}</h1>")
```

Comment:
This should likely return an error status code instead of 200 OK

Reply:
And probably no point in it being HTML instead of e.g. a JSON that can be reasonably parsed